### PR TITLE
feat: added generate extra api support

### PR DESCRIPTION
**Pull request by alvindaiyan**
## Description

added extra generate api only

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by NingLu** [comment]
We identified new issues on unchanged lines of code. Navigate to the Amazon CodeGuru Reviewer console to view the recommendations to fix them.

File: middleware_api/lambda/inference_v2/inference_api.py [hunk 1]
```diff
@@ -103,22 +69,62 @@ def prepare_inference(raw_event, context):
...
+                    # todo: need check if user has permission for the model
+                    if ckpt is None:
+                        ckpts_to_upload.append({
+                            'name': name,
+                            'ckpt_type': ckpt_type
+                        })
+                    else:
+                        ckpts.append(ckpt)
+
+            if len(ckpts_to_upload) > 0:
```
Comment (NingLu) [inline middleware_api/lambda/inference_v2/inference_api.py:102]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)

File: middleware_api/lambda/inference_v2/inference_api.py [hunk 2]
```diff
@@ -150,14 +156,17 @@ def run_inference(event, _):
...
+    models = {}
+    if 'used_models' in inference_job.params:
+        models = {
             "space_free_size": 4e10,
             **inference_job.params['used_models'],
         },
+
+    payload = InvocationsRequest(
+        task=inference_job.taskType,
+        username="test",
```
Comment (yike5460) [inline middleware_api/lambda/inference_v2/inference_api.py:168]: what does such username used for?

File: scripts/api.py [hunk 3]
```diff
@@ -195,7 +195,7 @@ def show_slim_dict(payload):
                 logger.info(f"waiting thread too much in condition pool {len(thread_deque)}, max: {CONDITION_POOL_MAX_COUNT}")
                 raise MemoryError
 
-            print(f'current version: dev')
+            print(f'current version: dev - extra generate')
```
Comment (yike5460) [inline scripts/api.py:198]: logger.info

File: middleware_api/lambda/inference_v2/test_inference_api.py [hunk 4]
```diff
@@ -174,3 +163,43 @@ def test_list_all_inference_jobs(self):
...
+        self._do_generate_extra('extra-batch-images', 'payload_extra_batch.json')
+
+    def test_generate_rembg(self):
+        self._do_generate_extra('rembg', 'payload_rembg.json')
+
+    def _do_generate_extra(self, _task_type, payload_url):
+        from inference_v2.inference_api import prepare_inference, run_inference
+
+        event = {
+            'user_id': 'admin',
```
Comment (yike5460) [inline middleware_api/lambda/inference_v2/test_inference_api.py:180]: consider to make admin as global var if it is our built-in role

File: middleware_api/lambda/inference_v2/inference_api.py (hunk 1 above)
Comment (yike5460) [inline middleware_api/lambda/inference_v2/inference_api.py:102]: according to the logic, we should ignore the ckpts.append operation and return error directly

**Review by yike5460 — COMMENTED** [review]
comments leaved

**Review by yike5460 — APPROVED** [review]

**Pull request merged**